src/Transport/ApiSnmp.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\PhpSnmp\Transport;
+
+use SimPod\PhpSnmp\Exception\SnmpApiError;
+use function array_key_exists;
+use function base64_decode;
+use function curl_close;
+use function curl_error;
+use function curl_init;
+use function is_array;
+use function is_bool;
+use function json_decode;
+use function json_last_error_msg;
+use function strtr;
+use const CURLOPT_CONNECTTIMEOUT;
+use const CURLOPT_HEADER;
+use const CURLOPT_RETURNTRANSFER;
+use const CURLOPT_TIMEOUT;
+use const JSON_BIGINT_AS_STRING;
+
+final class ApiSnmp implements Snmp
+{
+    private const API_PATH = '/snmp/{community}/{host}/{oid}';
+
+    /** @var string */
+    private $apiHostUrl;
+
+    /** @var string */
+    private $community;
+
+    /** @var string */
+    private $host;
+
+    /** @var int */
+    private $timeout;
+
+    public function __construct(
+        string $apiHostUrl,
+        string $community = 'public',
+        string $host = '127.0.0.1',
+        int $timeout = 30
+    ) {
+        $this->apiHostUrl = $apiHostUrl;
+        $this->community = $community;
+        $this->host = $host;
+        $this->timeout = $timeout;
+    }
+
+    /**
+     * @return iterable<string, mixed>
+     */
+    public function walk(string $oid) : iterable
+    {
+        yield from $this->getResponse($oid, true);
+    }
+
+    /**
+     * @return iterable<string, mixed>
+     */
+    public function walkWithCompleteOids(string $oid) : iterable
+    {
+        yield from $this->getResponse($oid, false);
+    }
+
+    private function getResponse(string $oid, bool $stripOidPrefix) : iterable
+    {
+        $url = $this->apiHostUrl . strtr(
+                self::API_PATH,
+                [
+                    '{community}' => $this->community,
+                    '{host}' => $this->host,
+                    '{oid}' => $oid,
+                ]
+            );
+
+        if ($stripOidPrefix) {
+            $url .= '?strip=1';
+        }
+
+        $curl = curl_init($url);
+        if (is_bool($curl)) {
```

Comment:
```suggestion
        if ($curl === false) {
```

Reply:
yeah its possible here, but usually phpstan has problems with this, because you check for false, but then it still can be true (if you define return type as eg `string|bool`), even though the implementation only returns false.

here we can use `=== false` because the phpdoc is wrong (claims to return type `false`, which ofc doesnt exists, but the tooling gets it lol)

Reply:
AFAIK it's common practice to define `false` in phpdocs return type instead of bool. PHPStan copes with it well (haven't found a case when it didn't work)

Reply:
Well yeah, it might be a common practice but it is fundamentally flawed :D phpstan manages it alright (it acknowledges the hack), but failes for cases with the whole boolean ... which makes sense... the problem is with returning two different data types and not checking the whole set of possible values ... phpstan does it's job well, but therefore we have to use is_bool instead of just direct check against `false` :)